core/src/blockchain/blockchain.rs

```diff
@@ -120,6 +136,7 @@ impl BlockChain {
         }
 
         assert!(self.pending_best_block_hash.read().is_none());
+        assert!(self.pending_highest_block_hash.read().is_none());
```

Comment:
What happend if `insert_block` called twice at the same time?

Reply:
The `importer` has an `import_lock` only one thread could import at a time. Since `insert_block` is called by `importer`, we don't need to consider the multithread situation.